main.js

```diff
@@ -96,34 +143,38 @@ function toggleTitleVisibility() {
 // Creates a grid from the inputed column and rows
 function createGridItems() {
     removeItems();
-    let numberOfColumns = columnInput.value;
-    let numberOfRow = rowInput.value;
-    let itemWidth =  window.innerWidth/numberOfColumns;
+    const numberOfColumns = columnInput.value;
+    const numberOfRows = rowInput.value;
+    const itemWidth =  window.innerWidth / numberOfColumns;
 
-    for (i=0; i<(numberOfColumns * numberOfRow); i++) {
-        addItem()
+    let index = 0
+    for (row = 0; row < (numberOfRows); row++) {
+        for (col = 0; col < (numberOfColumns); col++) {
+            addItem();
+        }
     }
-
-    //TODO: Math needs some work. Breaks down after
-    const items = document.querySelectorAll('.item');
-    items.forEach(item => {
-        let margin = 5;
-        let margins = (margin * 2) * (numberOfColumns - 1);
-        item.style.setProperty('width', ((itemWidth - margins) + "px"));
-     })
-
+    
      grid.refreshItems().layout();
 }
 
 // Add a grid item to DOM
 function addItem() {
-    var id = gridElement.children.length + 1;
-    var fragment = createDOMFragment(
+    const id = gridElement.children.length + 1;
+    let jobsDropDownHtml = createDropDownHTML('jobs', widgetsAndJobsList, (id - 1));
+    let widgetsDropDownList = createDropDownHTML('widgets', widgetsAndJobsList, (id - 1));
+    const fragment = createDOMFragment(
         '<div class="item">' + 
-            '<div class="item-content-default">' + id + '</div>' +
+            '<div class="item-content-default">' + 
+            id + 
```

Comment:
It looks like this might be the only place where id is used without subtracting 1. An alternative could be to add 1 to the id on this line and on line 162 use `const id = gridElement.children.length`. Then 1 wouldn't need to be subtracted from the other uses of id

Reply:
Yes, that makes the code cleaner, done.